setup_wordpress.sh

```diff
@@ -1,3 +1,245 @@
+sudo systemctl reload nginx
+
+echo "------------------ Creating Nginx Index ------------------"
+echo "
+<html>
+  <head>
+    <title>Rojin's Website</title>
+  </head>
+  <body>
+    <h1>Hello Rojin!</h1>
+  </body>
+</html>" | sudo tee /var/www/wordpress/index.html
+
+echo "You Can Test The Index Page At ${ip_address}"
+
+echo "------------------ Creating SSL Certificate ------------------"
+sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/nginx-selfsigned.key -out /etc/ssl/certs/nginx-selfsigned.crt
+sudo openssl dhparam -out /etc/nginx/dhparam.pem 2048
+
+echo "------------------ Creating Configuration Snippet ------------------"
+sudo echo "
+ssl_certificate /etc/ssl/certs/nginx-selfsigned.crt;
+ssl_certificate_key /etc/ssl/private/nginx-selfsigned.key;" | sudo tee /etc/nginx/snippets/self-signed.conf
+
+echo "------------------ Creating ssl-params.conf ------------------"
+sudo echo '
+ssl_protocols TLSv1.2;
+ssl_prefer_server_ciphers on;
+ssl_dhparam /etc/nginx/dhparam.pem;
+ssl_ciphers ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384;
+ssl_ecdh_curve secp384r1; # Requires nginx >= 1.1.0
+ssl_session_timeout  10m;
+ssl_session_cache shared:SSL:10m;
+ssl_session_tickets off; # Requires nginx >= 1.5.9
+# ssl_stapling on; # Requires nginx >= 1.3.7
+# ssl_stapling_verify on; # Requires nginx => 1.3.7
+resolver 8.8.8.8 8.8.4.4 valid=300s;
+resolver_timeout 5s;
+# Disable strict transport security for now. You can uncomment the following
+# line if you understand the implications.
+# add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
+add_header X-Frame-Options DENY;
+add_header X-Content-Type-Options nosniff;
+add_header X-XSS-Protection "1; mode=block";' | sudo tee /etc/nginx/snippets/ssl-params.conf
+
+echo "------------------ Adjusting Nginx To SSL ------------------"
+sudo cp /etc/nginx/sites-available/wordpress /etc/nginx/sites-available/wordpress.bak
+
+echo "
+server {
+    listen 443 ssl;
+    listen [::]:443 ssl;
+    include snippets/self-signed.conf;
+    include snippets/ssl-params.conf;
+
+    server_name $ip_address;
+
+    root /var/www/wordpress/html;
+    index index.html index.htm index.nginx-debian.html;
+
+}" | sudo tee /etc/nginx/sites-available/wordpress
+
+echo "
+
+server {
+    listen 80;
+    listen [::]:80;
+
+    server_name $ip_address;
+
+    return 301 https://\$server_name\$request_uri;
+}" | sudo tee /etc/nginx/sites-available/wordpress
+
+echo "------------------ Adjusting Firewall ------------------"
+sudo ufw allow 'Nginx Full'
+sudo ufw delete allow 'Nginx HTTP'
+
+echo "------------------ Testing Nginx ------------------"
+sudo nginx -t
+
+echo "------------------ Restarting Nginx ------------------"
+sudo systemctl restart nginx
+
+echo "------------------ Creating WordPress Database ------------------"
+read -p "Enter Your MYSQL Password: " mysql_password
```

Comment:
I think here could be set some static password. For example `mY$3qurePwaarodf`. In the future you could read it from secret file, or environment variable or from somewhere else.

Reply:
Great hint. Thanks